StoreKitUnitTests/ProductDetailsTests.swift

```diff
@@ -0,0 +1,115 @@
+            expect(sk1Product.price) == equivalentSK2Product.price
+            expect(sk1Product.localizedPriceString) == equivalentSK2Product.localizedPriceString
+            expect(sk1Product.productIdentifier) == equivalentSK2Product.productIdentifier
+            expect(sk1Product.isFamilyShareable) == equivalentSK2Product.isFamilyShareable
+            expect(sk1Product.localizedTitle) == equivalentSK2Product.localizedTitle
+            if sk1Product.subscriptionGroupIdentifier != nil {
+                expect(sk1Product.subscriptionGroupIdentifier) == equivalentSK2Product.subscriptionGroupIdentifier
+            } else {
+                expect(equivalentSK2Product.subscriptionGroupIdentifier).to(beNil())
+            }
+        }
+    }
+
+    func testSk1DetailsWrapsCorrectly() throws {
+        let productIdentifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        let sk1Fetcher = ProductsFetcherSK1(productsRequestFactory: ProductsRequestFactory())
+        var callbackCalled = false
+
+        sk1Fetcher.products(withIdentifiers: Set([productIdentifier])) { productDetailsSet in
+            callbackCalled = true
+            guard let productDetails = productDetailsSet.first else { fatalError("couldn't get product!") }
+
+            expect(productDetails.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
+            expect(productDetails.localizedDescription) == "Monthly subscription with a 1-week free trial"
+            expect(productDetails.price.description) == "4.99"
+            expect(productDetails.localizedPriceString) == "$4.99"
+            expect(productDetails.productIdentifier) == productIdentifier
+            expect(productDetails.isFamilyShareable) == true
+            expect(productDetails.localizedTitle) == "Monthly Free Trial"
+        }
+
+        expect(callbackCalled).toEventually(beTrue(), timeout: .seconds(5))
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func testSk2DetailsWrapsCorrectly() async throws {
+        guard #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) else { return }
+
+        let productIdentifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        let sk2Fetcher = ProductsFetcherSK2()
+
+        let productDetailsSet = try await sk2Fetcher.products(identifiers: Set([productIdentifier]))
+
+        let productDetails = try XCTUnwrap(productDetailsSet.first)
+
+        expect(productDetails.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
+        expect(productDetails.localizedDescription) == "Monthly subscription with a 1-week free trial"
+        expect(productDetails.price.description) == "4.99"
+        expect(productDetails.localizedPriceString) == "$4.99"
+        expect(productDetails.productIdentifier) == productIdentifier
+        expect(productDetails.isFamilyShareable) == true
+        expect(productDetails.localizedTitle) == "Monthly Free Trial"
+
```

Comment:
Extra space here, also, this test and the previous is not testing for `subscriptionGroupIdentifier`

Reply:
wow that's a great catch, totally missed it

Reply:
here's an interesting thing: 
the `subscriptionGroupIdentifier`'s value isn't the reference name (i.e.: the display name, in this case "subscription_group", it's the ID:
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/3922667/134724893-f8d1d9a2-7f80-4481-9d6b-1c402b1e99c0.png">

With that being the case, I'm not sure it's worth checking it directly, but happy to add it. I'll check if it's visible in the StoreKit Config file at least

Reply:
<img width="361" alt="image" src="https://user-images.githubusercontent.com/3922667/134724983-6214e102-dcf4-41bd-ad53-3de6b7ce9261.png">
okay, it's visible there as long as you edit as JSON, so I think it makes sense to check it